models/stg_iterable__user_unsubscribed_channel.sql

```diff
@@ -15,23 +15,34 @@ fields as (
         */
         {{
             fivetran_utils.fill_staging_columns(
-                source_columns=adapter.get_columns_in_relation(ref('stg_iterable__user_unsubscribed_channel_history_tmp')),
-                staging_columns=get_user_unsubscribed_channel_history_columns()
+                source_columns=adapter.get_columns_in_relation(ref('stg_iterable__user_unsubscribed_channel_tmp')),
+                staging_columns=get_user_unsubscribed_channel_columns()
             )
         }}
         
     from base
 ),
 
 final as (
-    
-    select 
-        channel_id,
-        email,
-        updated_at,
+
+    select
+
+        cast(_fivetran_id as {{ dbt.type_string() }} ) as _fivetran_user_id,
+        coalesce(_fivetran_id, email) as unique_user_key,
+        cast(channel_id as {{ dbt.type_string() }} ) as channel_id,
+        {{ dbt_utils.generate_surrogate_key(['_fivetran_id', 'channel_id', 'email', 'updated_at']) }} as unsub_channel_unique_key,
         _fivetran_synced
+
     from fields
 )
 
-select * 
+
+select 
+    *
+
+{% if does_table_exist('user_unsubscribed_channel') == 'false' %}
```

Comment:
Resorting to this because when trying to do the coalesce method in the final CTE, we run into a redshift error in both the `partition by` and the `order by`  when it's a null field. The coalesce doesn't work for the `order_by` because the 2 fields need to be the same data type

` No matching signature for function COALESCE for argument types: TIMESTAMP, STRING.` 

And there's not another timestamp field.